src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -33,11 +36,40 @@ public class CoraxDocumentConverter : CoraxDocumentConverterBase
         object value;
         foreach (var indexField in _fields.Values)
         {
-            if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, indexField.OriginalName ?? indexField.Name, out value))
+            if (indexField.Spatial is AutoSpatialOptions spatialOptions)
+            {
+                var spatialField = CurrentIndexingScope.Current.GetOrCreateSpatialField(indexField.Name);
+
+                switch (spatialOptions.MethodType)
+                {
+                    case AutoSpatialOptions.AutoSpatialMethodType.Wkt:
+                        if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, spatialOptions.MethodArguments[0], out var wktValue) == false)
+                            continue;
+
+                        value = StaticIndexBase.CreateSpatialField(spatialField, wktValue);
+                        break;
+                    case AutoSpatialOptions.AutoSpatialMethodType.Point:
+                        if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, spatialOptions.MethodArguments[0], out var latValue) == false)
+                            continue;
+
+                        if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, spatialOptions.MethodArguments[1], out var lngValue) == false)
+                            continue;
+
+                        value = StaticIndexBase.CreateSpatialField(spatialField, latValue, lngValue);
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
```

Comment:
Please pass `spatialOptions.MethodType` to the exception

Reply:
done